src/eckit/geometry/Sphere.cc

```diff
@@ -197,9 +190,23 @@ void Sphere::convertSphericalToCartesian(const double& radius, const Point2& Alo
      *
      * These three conditionings combined project very accurately to the sphere
      * poles and quadrants.
+     *
+     * Latitudes outside the standard interval [-90°,90°] are first normalised
+     * into the interval [-90°,270°], then any points with latitudes in
+     * [90°,270°] are flagged as "across the pole". Such points are re-labeled
+     * with equivalent coordinates that lie within the standard coordinate patch
+     * by the transformation
+     *   (λ, ϕ) -> (λ+180°, 180°-ϕ)
+     * As the convertSphericalToCartesian algorithm depends on sin(ϕ) which is
+     * invariant under this transformation, the normalisation procedure is
+     * simplified to only perform the longitude phase shift.
      */
 
-    const double lambda_deg = normalise_longitude(Alonlat[0], -180.);
+    // We normalise the latitude by calling the logically-identical normalise_longitude.
+    const auto& normalise_latitude = normalise_longitude;
+    const bool lat_across_pole = (normalise_latitude(Alonlat[1], -90.) > 90.);
+
+    const double lambda_deg = normalise_longitude(Alonlat[0] + (lat_across_pole ? 180. : 0.), -180.);
```

Comment:
Hi - Maybe I am missing something but I can't quite see how you are dealing with the South Pole.

Reply:
```
const bool lat_across_pole = (normalise_latitude(Alonlat[1], -90.) > 90.);
```
does not apply normalisation to the used latitude. It just checks if values are between -90 and +90.
`normalise_latitude(Alonlat[1], -90.)` -> always value between -90. and 270.
So I'm wondering, how is this different from
```
const bool lat_across_pole=(Alonlat[1] > 90. || Alonlat[1] < -90.);
```

Reply:
If the input latitude has sufficiently large values that it "wraps around the poles" a few times, then the two are different. This is because a point can have very large latitudes still be in the same hemisphere. As an extreme example, consider `(lon,lat) = (x,370)`; this point is also `(x,10)`, in the same hemisphere, and the coordinates can be "fixed" without needing to shift the longitude.

In this example...
- `(Alonlat[1] > 90. || Alonlat[1] < -90.)` would evaluate to true, even though we do not want to shift the longitude value
- `normalise_latitude(Alonlat[1], -90.)` returns 10, which is in [-90,90], consistent with not editing the longitude

Does that help? Or have I missed something?

Reply:
@MarekWlasak the statement `normalise_latitude(Alonlat[1], -90.)` means that points "below the south pole" are re-labeled with a latitude "above the north pole"... e.g. `lat=-100 --> lat=260`. Using the `normalise_latitude` function avoids writing complicated branching to achieve the desired behavior.

Reply:
Do I need to write more documentation? :)

Reply:
@MarekWlasak can you clarify what discontinuity you're referring to? This is just choosing a consistent "branch cut" of the periodic coordinates before converting to cartesian coordinates... I don't expect anything will be discontinuous in the output space...